python/hidet/graph/ops/definitions/reduce/reduce_f16.py

```diff
@@ -293,6 +293,12 @@ def __init__(self, x: Tensor, dims: Optional[Sequence[int]], keepdims: bool = Fa
 def reduce_f16(
     x: Tensor, dims: Union[int, Sequence[int]], keepdims: bool, reduce_type: Union[ReduceType, str]
 ) -> Tensor:
+    from hidet.ir import Expr
+
+    if True in [isinstance(i, Expr) for i in x.shape]:
```

Comment:
Personally, I prefer 
```suggestion
    if any(isinstance(d, Expr) for d in x.shape):
```